src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -1037,8 +1039,15 @@ void Hole::updateHoleCutParams()
             if (HoleCutCountersinkAngle.getValue() == 0.0) {
                 HoleCutCountersinkAngle.setValue(getCountersinkAngle());
             }
-            if (HoleCutDepth.getValue() == 0.0 && holeCutTypeStr == "Counterdrill") {
-                HoleCutDepth.setValue(1.0);
+            if (HoleCutDepth.getValue() == 0.0) {
+                if (holeCutTypeStr == "Counterdrill") {
+                    HoleCutDepth.setValue(1.0);
+                } else {
+                    HoleCutDepth.setValue(
+                        (HoleCutDiameter.getValue() / 2)
+                        / std::tan(HoleCutCountersinkAngle.getValue() / 2)

```

Comment:
This can lead to division by zero.  Can we guard against that somehow?

In ui it could be done by specifying the range of the input, but for raw input (or if expressions are used) we need to figure out something else.